oauth2client/contrib/gce.py

```diff
@@ -125,32 +257,15 @@ def _refresh(self, http_request):
         Raises:
             HttpAccessTokenRefreshError: When the refresh fails.
         """
-        response, content = http_request(
-            META, headers={'Metadata-Flavor': 'Google'})
-        content = _from_bytes(content)
-        if response.status == http_client.OK:
-            try:
-                token_content = json.loads(content)
-            except Exception as e:
-                raise HttpAccessTokenRefreshError(str(e),
-                                                  status=response.status)
-            self.access_token = token_content['access_token']
-        else:
-            if response.status == http_client.NOT_FOUND:
-                content += (' This can occur if a VM was created'
-                            ' with no service account or scopes.')
-            raise HttpAccessTokenRefreshError(content, status=response.status)
-
-    @property
-    def serialization_data(self):
-        raise NotImplementedError(
-            'Cannot serialize credentials for GCE service accounts.')
-
-    def create_scoped_required(self):
-        return False
+        self.access_token, self.token_expiry = _get_access_token(
+            http_request,
+            self._service_account_info.get('email', 'default'))
 
     def create_scoped(self, scopes):
-        return AppAssertionCredentials(scopes, **self.kwargs)
+        # Trigger warning or error based on scopes
+        self.scopes = scopes
```

Comment:
Avoid calling one part of the public API of an object from within the implementation of another part of the public API of the same object.

Why is `self.scopes = scopes` better here than `self._check_scopes_and_notify(scopes)`? If you chose the latter the purpose would be clear and you could drop the implementation comment, right?